lms/templates/video.html

```diff
@@ -52,36 +54,22 @@ <h3 class="hd hd-4 downloads-heading sr" id="video-download-transcripts_${id}">$
         % if download_video_link or public_video_url:
             <div class="wrapper-download-video">
                 <h4 class="hd hd-5">${_('Video')}</h4>
-                % if public_video_url:
-                    <span class="icon fa fa-share-alt" aria-hidden="true"></span>
-                    ${_('Share on:')}
-                    <a
-                        class="btn-link"
-                        id="twitter-share-button"
-                        href="${public_video_url}"
-                        target="_blank"
-                    >
-                        <span class="icon fa fa-twitter-square" aria-hidden="true"></span>
-                        <span class="sr">${_('Share on Twitter')}</span>
-                    </a>
-                    <a
-                        class="btn-link"
-                        id="facebook-share-button"
-                        href="${public_video_url}"
-                        target="_blank"
-                    >
-                        <span class="icon fa fa-facebook-square" aria-hidden="true"></span>
-                        <span class="sr">${_('Share on Facebook')}</span>
-                    </a>
-                    <a
-                        class="btn-link"
-                        id="linkedin-share-button"
-                        href="${public_video_url}"
-                        target="_blank"
-                    >
-                        <span class="icon fa fa-linkedin-square" aria-hidden="true"></span>
-                        <span class="sr">${_('Share on LinkedIn')}</span>
-                    </a>
+                % if public_video_url and social_share_sites:
+                    <div class="wrapper-social-share" data-url="${public_video_url}">
+                        <span class="icon fa fa-share-alt" aria-hidden="true"></span>
+                        ${_('Share on:')}
+                        % for social_share_site in social_share_sites:
+                            <a
```

Comment:
I don't know I like this.
I understand why this helps refactor. But the result would be that the user cannot right click on this link and open the sharing window in a new tab, right? 
Also, are you concerned about accessibility impact of this?

Reply:
That's technically not new behavior with this refactor. Current behavior if you open one of these in a new tab is that the courses.edx.org/video/<block_id> window opens in a new tab, which I don't think is necessarily "correct". Now it will open the current page again in a new tab. That's also not "right" but I'm not sure what a "correct" behavior would be. I'm completely open to changes to resolve accessibility concerns though

Reply:
This is resolved by using the new "sharing_url" href